java/org/apache/coyote/Request.java

```diff
@@ -853,24 +846,22 @@ public boolean isProcessing() {
      * @param contentType a content type header
      */
     private static String getCharsetFromContentType(String contentType) {
-
         if (contentType == null) {
             return null;
         }
         int start = contentType.indexOf("charset=");
         if (start < 0) {
             return null;
         }
-        String encoding = contentType.substring(start + 8);
-        int end = encoding.indexOf(';');
+        start += 8;
+        int end = contentType.indexOf(';', start);
+        String encoding;
         if (end >= 0) {
-            encoding = encoding.substring(0, end);
+            encoding = contentType.substring(start, end).trim();
+        } else {
+            encoding = contentType.substring(start).trim();
         }
-        encoding = encoding.trim();
-        if ((encoding.length() > 2) && (encoding.startsWith("\"")) && (encoding.endsWith("\""))) {
-            encoding = encoding.substring(1, encoding.length() - 1);
-        }
-
-        return encoding.trim();
+        return encoding.replaceAll("\"", "");
```

Comment:
This changes functionality. Granted the existing code doesn't strictly implement the requirements of RFC 9110 either but "code clean-up" PRs should not be changing functionality without explicitly drawing attention to that fact.

Separately, I have concerns about the performance implications of this change. Srtring.replaceAll() is not cheap as it uses regular expressions.

If we want to specification compliant here (and we probably should be) the right way to do this would be to use  o.a.t.u.http.parser.MediaType.

Reply:
I'm curious whether there are any performance advantages when using MediaType to comply with the specification compared to the existing code. Additionally, I'm interested in knowing whether the existing code already adhered to the specification as well as MediaType does.